adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -339,6 +339,373 @@ private[adam] class VariantContextConverter extends Serializable with Logging {
+    if (adr.size > (index + 1)) {
+      vab.setReverseReadDepth(adr.get(index + 1).asInstanceOf[java.lang.Integer])
+    }
+    vab
+  }
+
+  private[converters] def formatTranscriptEffects(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    TranscriptEffectConverter.convertToTranscriptEffects(v, vc).foreach(vab.setTranscriptEffects(_))
+    vab
+  }
+
+  private val variantAnnotationConversionFns: Iterable[(HtsjdkVariantContext, VariantAnnotation.Builder, Variant, Int) => VariantAnnotation.Builder] = Iterable(
+    formatAncestralAllele(_, _, _, _),
+    formatDbSnp(_, _, _, _),
+    formatHapMap2(_, _, _, _),
+    formatHapMap3(_, _, _, _),
+    formatThousandGenomes(_, _, _, _),
+    formatCigar(_, _, _, _),
+    formatAlleleCount(_, _, _, _),
+    formatAlleleFrequency(_, _, _, _),
+    formatReadDepth(_, _, _, _),
+    formatForwardReadDepth(_, _, _, _),
+    formatReverseReadDepth(_, _, _, _),
+    formatTranscriptEffects(_, _, _, _)
+  )
+
+  private[converters] def extractAncestralAllele(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getAncestralAllele).foreach(vcb.attribute("AA", _))
+    vcb
+  }
+
+  private[converters] def extractDbSnp(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getDbSnp).foreach(vcb.attribute("DB", _))
+    vcb
+  }
+
+  private[converters] def extractHapMap2(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getHapMap2).foreach(vcb.attribute("H2", _))
+    vcb
+  }
+
+  private[converters] def extractHapMap3(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getHapMap3).foreach(vcb.attribute("H3", _))
+    vcb
+  }
+
+  private[converters] def extractThousandGenomes(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getThousandGenomes).foreach(vcb.attribute("1000G", _))
+    vcb
+  }
+
+  private[converters] def extractCigar(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getCigar).foreach(vcb.attribute("CIGAR", _))
+    vcb
+  }
+
+  private[converters] def extractAlleleCount(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getAlleleCount).foreach(vcb.attribute("AC", _))
+    vcb
+  }
+
+  private[converters] def extractAlleleFrequency(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getAlleleCount).foreach(vcb.attribute("AF", _))
+    vcb
+  }
+
+  private[converters] def extractReadDepth(
+    va: VariantAnnotation,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    Option(va.getReadDepth).foreach(rd => vcb.attribute("AD", prependDefaultReferenceValue(rd)))
```

Comment:
You've got a lot of long `Option(va.get<Fieldname>).foreach(` lines here and in the next few methods. Mind wrapping them?